Generic Field Type tutorial (#733)
ezsystems/developer-documentation: docs/tutorials/generic_field_type/2_define_point2d_field_type.md

# Step 2 - Define the Point 2D Field Type

!!! tip

    You can find all files used and modified in this step on [GitHub]().

## The Type class

The Type contains logic of the Field Type: validating data, transforming from various formats, describing the validators, etc.

> In our example Point 2D Field Type will extend `eZ\Publish\SPI\FieldType\Generic\Type` class.

Reviewer: ```suggestion
In this example Point 2D Field Type will extend the `eZ\Publish\SPI\FieldType\Generic\Type` class.
```

# Step 2 - Define the Point 2D Field Type

!!! tip

    You can find all files used and modified in this step on [GitHub]().

## The Type class

The Type contains logic of the Field Type: validating data, transforming from various formats, describing the validators, etc.
A Generic Type class must implement `eZ\Publish\SPI\FieldType\Generic\Type` (Generic Field Type interface).

> For more information about the Type class of a Field Type see [Type class](../../api/field_type_type_and_value.md#type-class).

Reviewer 2: ```suggestion
For more information about the Type class of a Field Type, see [Type class](../../api/field_type_type_and_value.md#type-class).
```

## Field Type identifier

> First create `src/FieldType/Point2D/Type.php`.

Reviewer 2: ```suggestion
First, create `src/FieldType/Point2D/Type.php`.
```
